money.go

```diff
@@ -282,6 +282,12 @@ func (m *Money) Allocate(rs ...int) ([]*Money, error) {
 		total += party.amount
 	}
 
+	// if the sum of all ratios is zero, then we just returns zeros and don't do anything
+	// with the leftover
+	if sum == 0 {
```

Comment:
would it work with the case when ratios are equal to {-50, 50}? The sum is `0`, how this works? To be honest with you, I'm not really sure what negative ratio means and I expect to always get unsigned integer, is there a use case I'm missing?

Reply:
I couldn't come up with a use case for negative ratios. I've added a check, that returns a new error if any of the ratios are negative.